plugins/modules/source_control/gitlab/gitlab_runner.py

```diff
@@ -246,18 +260,28 @@ def updateRunner(self, runner, arguments):
     '''
     @param description Description of the runner
     '''
-    def findRunner(self, description):
-        runners = self._gitlab.runners.all(as_list=False)
+    def findRunner(self, description, owned=False):
+        if owned:
+            runners = self._gitlab.runners.list(as_list=False)
+        else:
+            runners = self._gitlab.runners.all(as_list=False)
+
         for runner in runners:
-            if (runner['description'] == description):
-                return self._gitlab.runners.get(runner['id'])
+            # python-gitlab 2.2 through at least 2.5 returns a list of dicts for list() instead of a Runner
+            # object, so we need to handle both, getattr allows that
```

Comment:
`getattr allows that` - you mean `hasattr` ?

Reply:
```suggestion
            # object, so we need to handle both, hasattr allows that
```